docs/enhancements/ipxe-host-boot.md

```diff
@@ -0,0 +1,125 @@
+provide an API to download the boot script. The boot script created and served 
+by the assisted service will contain URLs to the required artifacts hosted by 
+the image service.
+
+### User Stories
+
+#### Story 1
+
+As a user of assisted installer I want to boot my hosts using iPXE without 
+setting up my own http server to host artifacts.
+
+### Implementation Details/Notes/Constraints
+
+#### Ironic Integration
+
+To support integration with Ironic’s existing iPXE boot methods, the 
+discovery ignition must be appended to the initrd rather than specified as 
+a separate kernel parameter. The image service will also handle this initrd 
+manipulation when an initrd is requested for a particular infra-env.
+
+#### New Image Service APIs
+
+Image service APIs will be added to download the required boot artifacts. The 
+rootfs and kernel will be served directly from the template image. The initrd 
+endpoint will be specific to an infra-env in order to embed the ignition.
+
+- `GET /boot-artifacts/{artifact-name}?{params}`
+  - artifact-name can be either “rootfs” or “kernel”
+  - params must include “version” and “arch” which are used to identify a template image
+  - No authentication is required as we’re serving these directly from the uncustomized image
+  - Examples
+    - `/boot-artifacts/rootfs?version=4.9&arch=x86_64`
+    - `/boot-artifacts/kernel?version=4.10&arch=arm64`
+- `GET /images/{image-id}/pxe-initrd?{params}`
+  - image-id is the uuid of an infra-env in assisted-service
+  - params must include “version” and “arch”
+  - params may include “api_key” or “image_token” depending on assisted-service authentication configuration
+  - Example
+    - `/images/afe293b9-e3a0-47ed-8952-f26721626497/pxe-initrd?version=4.10&arch=x86_64&image_token={token}`
+
+#### New Assisted Service REST APIs
+
+Assisted service APIs will be added to download the ipxe boot script and to 
+retrieve a presigned url for downloading the boot script.
+
+- `GET /v2/infra-envs/{infra-env-id}/downloads/ipxe-script`
+  - Will return an iPXE boot script with artifact URLs pointing to the image service
+  - Initrd URL will be presigned
+- `GET  /v2/infra-envs/{infra-env-id}/downloads/ipxe-script-url`
+  - Will return a presigned URL for the iPXE script
+
+#### Assisted Service Kubernetes APIs
+
+A presigned download url for the iPXE boot script will be added to the infra-env 
+resource status. This URL will be set as a part of the infra-env controller reconcile loop.
+
+### Risks and Mitigations
+
+## Design Details [optional]
+
+### Open Questions
+
+When running in the cloud how do we set expiration times for the tokens we use for 
+auth when downloading the boot script and the initrd?
+The boot script can probably be treated similarly to the image as it is only downloaded once.
+The initrd will be downloaded every time a machine boots and the url needs to remain valid.
+This raises UX-type questions. Will people point their machines at the boot script endpoint 
```

Comment:
hmm looking into equinix option, they let u provide script endpotint. I would assume it can be pointed directly to the SaaS. Otherwise, someone will have to host it - more resource, extra machine, networking. I would like to avoid it. @javipolo wdyt?

Reply:
I'd totally point directly to the SaaS, it makes all the process way easier and cleaner from the user point of view